backend/route/classdata_test.go

```diff
@@ -0,0 +1,190 @@
+package route
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"regexp"
+	"sort"
+	"testing"
+
+	"github.com/DATA-DOG/go-sqlmock"
+
+	"github.com/ucsdscheduleplanner/UCSD-Schedule-Planner/backend/store"
+)
+
+var getClassDataColumns = []string{
+	"Department",
+	"CourseNum",
+	"SectionId",
+	"CourseId",
+	"ClassType",
+	"Days",
+	"Time",
+	"Location",
+	"Room",
+	"Instructor",
+	"Description"}
+
+func TestBuildQueryClassData(t *testing.T) {
+	testRequests := [][]ClassDataRequest{
+		{{"SP19", "LTEN", "25"}},
+		{{"SP19", "LTEN", "26"}, {"SP19", "LTEN", "26"}},
+		{{"SP19", "LTEN", "26"}, {"FA20", "LTEN", "26"}},
+	}
+
+	// note: there is no order of quarters and queryParams guaranteed
+	// use same class to make testing easier
+
+	expects := []struct {
+		queryRex    string
+		quarters    []string
+		queryParams []interface{}
+	}{
+		{
+			fmt.Sprintf(
+				`SELECT %s FROM %s WHERE \(DEPARTMENT, COURSE_NUM\) IN \(\(\?,\?\)\)`, classDataColumns, "SP19",
+			),
+			[]string{"SP19"},
+			[]interface{}{"LTEN", "25"},
+		},
+		{
+			fmt.Sprintf(
+				`SELECT %s FROM %s WHERE \(DEPARTMENT, COURSE_NUM\) IN \(\(\?,\?\),\(\?,\?\)\)`, classDataColumns, "SP19",
+			),
+			[]string{"SP19"},
+			[]interface{}{"LTEN", "26", "LTEN", "26"},
+		},
+		{
+			fmt.Sprintf(
```

Comment:
I see that you are combining all statements together instead of making separate statements. This is probably for a performance increase, but have you noticed any markably different performance differences? If the code is must more reasonable doing each query separately and you don't have full confidence in your method of doing this, then I would say the performance drop is justified. However, if you are confident then I'm willing to go along with it.